### DISCUSSION TITLE
ESC to exit Co-pilot suggestion

### DISCUSSION THREAD

**Discussion by pyronaur** [community/community#47070] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Up until the recent VSCode update, I was able to hit ESC to hide the Co-Pilot suggestion, it's no longer possible. Can we please get that functionality back? It's disrupting the workflow now. I can navigate suggestions with cmd + right left arrows, which is great, but I need a way other than typing something else to exit the suggestion.

**Comment by Juanescacha** [comment, marked as answer]
Hi, I have VSCode Updated also ando have been using Github Copilot for the last months and just recently check and the ESC functionality to hide or skip the Copilot  is STILL working as usual, try looking for any extension that maybe conflicts with the ESC key an overrides the short-cut or try re installing the extension or vscode.

**Reply by pyronaur** [reply]
That's weird. Can you install Co-pilot nightly to see if that changes anything?

**Reply by Juanescacha** [reply]
bro that's probably the issue, nightly version is a more updated version but sometimes they have bugs, not suitable for average users, etc.

Use the stable version, for not having this weird issues instead, or there is a reason that you need to use this nightly version instead?

**Reply by pyronaur** [reply]
Nightly hasn't had issues for me for over 6 months now, I hope that it's truly a bug and not a new feature.

**Reply by Juanescacha** [reply]
Just tested in a new profile, only with Copilot Nightly and the ESC functionality to exit Copilot suggestions worked the same as the stable version, so it's not a bug or new function, its possible that another extensions its getting conflicts or the extension config got mess up.

Why don't you try to create a new VSCODE profile to test the extension alone as I did, its pretty simple if you don't know:
![image](https://user-images.githubusercontent.com/6394715/218481814-c0e6e960-9b34-444b-a73f-a66e6837fbc2.png)
Hopes you figure it out, tells me if it helps or if dont't for me to try more things!!

**Reply by pyronaur** [reply]
<img width="317" alt="CleanShot 2023-02-13 at 21 32 00@2x" src="https://user-images.githubusercontent.com/988095/218555795-d57fd52b-a04c-4f3e-9807-6c44b1d9283a.png">

Ran the bisect twice, nothing found. Are you running Version: 1.75.0 ?

**Reply by Juanescacha** [reply]
![image](https://user-images.githubusercontent.com/6394715/218556627-c8d34ca3-6444-4526-91f8-8b3746ea9bc7.png)

you tried the profiles thing?

**Reply by pyronaur** [reply]
Are you seeing this Co-pilot inline suggestion style?
<img width="374" alt="CleanShot 2023-02-13 at 21 35 51@2x" src="https://user-images.githubusercontent.com/988095/218556812-ac48e546-836c-4407-99f9-134027bf33af.png">

**Reply by Juanescacha** [reply]
Yes

**Reply by pyronaur** [reply]
Weird, I activated a new profile, I can ESC, but I'm not seeing the toolbar for suggestions anymore.

**Reply by Juanescacha** [reply]
in my experience that toolbar appears only when you hover the mouse on it, 70% of the times I don't see that toolbar but I don't care cause the keyboard shortcuts there are the same, so I already memorized them :D

**Reply by Juanescacha** [reply]
it seems that its something on your main profile maybe a setting in the json or another extension that makes not possible the ESC, but as you can see on  a new profile with only copilot you can perfectly ESC

**Reply by pyronaur** [reply]
Ah, found it. Thanks for pointing me to the right path. It was this in my keyboard shortcuts:

```
{
		"key": "escape",
		"command": "-editor.action.inlineSuggest.hide",
		"when": "inlineSuggestionVisible"
	},
```

I added that a while back because my neovim is bound to cmd+esc, but I can't remember why this specific one was needed. Removed and now it works.

**Reply by Juanescacha** [reply]
Excellent @pyronaur , I'm glad that you figured it out!!!
don't forget to mark it as solved to help future developers with same issue 🤗
